syntax = "proto3";

package trader;

option go_package = ".;trader";

service Trader {
    rpc CreateMarketOrder (CreateMarketOrderRequest) returns (CreateMarketOrderResponse) {}
    rpc CreateLimitOrder (CreateLimitOrderRequest) returns (CreateLimitOrderResponse) {}
}

message CreateMarketOrderRequest {
    string opType = 1;
    string ticker = 2;
    int32 qty = 3;
}

message CreateMarketOrderResponse {}

message CreateLimitOrderRequest {
    string opType = 1;
    string ticker = 2;
    float price = 3;
    int32 qty = 4;
}

message CreateLimitOrderResponse {}
